### PR TITLE
Update io.fits_column._scalar_to_format to use min_scalar_type

### LINKED ISSUE #4838: Use min_scalar_type() in io.fits.column._scalar_to_format
There is a TODO note in ``astropy/io/fits/column.py`` that says:

```
    # TODO: Numpy 1.6 and up has a min_scalar_type() function that can handle
    # this; in the meantime we have to use our own implementation (which for
    # now is pretty naive)
```

Since we now support only Numpy 1.7+, this can be done.

### PR DESCRIPTION
**Pull request by simongibbons**
Does what it says on the tin.

Fixes #4838

### CODE REVIEW

**Comment by simongibbons** [comment]
Huh, nothing in the logs to suggest why the coverage build on Travis failed.

**Comment by bsipocz** [comment]
@simongibbons - Simple timeout, someone with commit rights should restart the build.

**Comment by pllim** [comment]
I don't see anything controversial here, so I am tagging 1.2.

@simongibbons , perhaps a change log entry under "API change" for 1.2.

@embray , do you wish to give a final blessing?

**Comment by simongibbons** [comment]
I'm happy to add one, but my understanding is that functions prefixed with an underscore aren't public so this shouldn't touch the public API.

**Comment by pllim** [comment]
Hmm. I was thinking along the line of "blahblahblah logic in FITS now requires Numpy 1.6 or later" but it might be redundant given #4784. Never mind if it is indeed unnecessary.

**Comment by embray** [comment]
Yeah, Astropy already requires Numpy >= 1.6 anyways.  This is just a minor refactoring as I see it. :+1:

File: astropy/io/fits/column.py [hunk 1]
```diff
@@ -2088,22 +2090,14 @@ def _scalar_to_format(value):
...
-                return char
-    elif isinstance(value, float):
-        # For now just assume double precision
-        return 'D'
-    elif isinstance(value, complex):
-        return 'M'
-    else:
-        return 'A' + str(len(value))
+    numpy_dtype_str = np.min_scalar_type(value).str
+    numpy_dtype_str = numpy_dtype_str[1:] #Strip endianness
```
Comment (embray) [inline astropy/io/fits/column.py]: This is a terrible nitpick but there should be two spaces before the `#` and one space after.

**Comment by simongibbons** [comment]
Cool that's the tiny formatting change to the comment done. I've squashed the commit down already because otherwise it's just noise!

**Comment by pllim** [comment]
Since @embray gave a green light, I am merging this. Thanks!

**Pull request merged**